components/pages/workPageLayout/WorkPageLayout.tsx

```diff
@@ -15,36 +15,42 @@ import {
   useGetMaterialQuery,
 } from "@/lib/graphql/generated/fbi/graphql"
 
+import { filterManifestationsByEdition, filterManifestationsByMaterialType } from "./helper"
+
 function WorkPageLayout({ workId }: { workId: string }) {
   const { data, isLoading } = useGetMaterialQuery({
     wid: workId,
   })
-  const work = data?.work ? (data.work as WorkFullWorkPageFragment) : null
 
   const [selectedManifestation, setSelectedManifestation] =
     useState<ManifestationWorkPageFragment>()
   const searchParams = useSearchParams()
 
+  if (!data || !data.work) {
+    notFound()
+  }
+
+  const work = data?.work as WorkFullWorkPageFragment
+  const manifestations = work?.manifestations.all as ManifestationWorkPageFragment[]
+
+  // Filter out manifestations that not digital except physical books
+  const filteredManifestationsByMaterialType = filterManifestationsByMaterialType(manifestations)
+  // If multiple manifestations has the same type find the latest edition
+  const filteredManifestationsByEdition = filterManifestationsByEdition(
+    filteredManifestationsByMaterialType
+  )
```

Comment:
While I dont see any programmatic problems here, I think there is too much logic piling up.
Would be nice to separate functionality out in functions or hooks.
Another think we could consider was to use function composition, like described [here](https://medium.com/@karimdhrif4444/understanding-function-composition-in-javascript-2f0168d685ac) (and loads of other articles :))
And that way we could do, something like:
```js
const processManifestations = composer(filterManifestationsByMaterialType, filteredManifestationsByMaterialType)

const workManifestations = processManifestations(manifestations)